components/TransactionList/index.js

```diff
@@ -1,13 +1,73 @@
+import { Fragment } from "react";
 import TransactionCard from "../TransactionCard";
-import { Ul } from "./TransactionList.styled";
+import * as Styled from "./TransactionList.styled";
+import {
+  findYearsInTransactions,
+  filterTransactionsByYear,
+  filterTransactionsByMonth,
+  sortMonths,
+} from "../../utils/transaction";
+import { months } from "@/utils/months";
 
 function TransactionList({ transactions }) {
+  const yearsInTransactions = findYearsInTransactions(transactions);
+
+  const transactionsPerYearsAndMonths = yearsInTransactions.map(
+    (year, index) => {
+      const transactionsPerYear = filterTransactionsByYear(transactions, year);
+      if (transactionsPerYear.length === 0) {
+        return null;
+      }
+
+      const monthsToMap =
+        index === 0
+          ? sortMonths(new Date().getMonth())
+          : months.slice().reverse();
+
+      const transactionsPerMonths = monthsToMap.map((month) => {
+        const monthIndex = months.indexOf(month);
+        const transactionsPerMonth = filterTransactionsByMonth(
+          transactionsPerYear,
+          monthIndex
+        );
+        if (transactionsPerMonth.length === 0) {
+          return null;
```

Comment:
here you return a different structure than in the final case. Depending on your future code this can be difficult to deal with. Another approach would be to return the same object but with an empty transactions array: 

```js
return { month, transactions: [] }

```
this is purely optional, but it might make it easier to deal with this data later on.